test/system/src/test/java/io/pravega/test/system/ReadTxnWriteScaleWithFailoverTest.java

```diff
@@ -73,35 +77,50 @@
     public static void initialize() throws InterruptedException, MarathonException, URISyntaxException {
         URI zkUri = startZookeeperInstance();
         startBookkeeperInstances(zkUri);
+        URI hdfsURI = startHDFSInstances();
         URI controllerUri = startPravegaControllerInstances(zkUri);
-        startPravegaSegmentStoreInstances(zkUri, controllerUri);
+        startPravegaSegmentStoreInstances(zkUri, controllerUri, hdfsURI);
     }
 
 
     @Before
     public void setup() {
         // Get zk details to verify if controller, segmentstore are running
-        Service zkService = new ZookeeperService("zookeeper");
+        Service zkService =  Utils.isDockerLocalExecEnabled() ? new ZookeeperDockerService("zookeeper") :
+                 new ZookeeperService("zookeeper");
         List<URI> zkUris = zkService.getServiceDetails();
         log.debug("Zookeeper service details: {}", zkUris);
         //get the zk ip details and pass it to  host, controller
         URI zkUri = zkUris.get(0);
 
+        URI hdfsUri = null;
+        if (Utils.isDockerLocalExecEnabled()) {
+            Service hdfsService = new HDFSDockerService("hdfs");
+            if (!hdfsService.isRunning()) {
+                hdfsService.start(true);
+            }
+            hdfsUri = hdfsService.getServiceDetails().get(0);
+            log.debug("HDFS service details: {}", hdfsService.getServiceDetails());
+        }
         // Verify controller is running.
-        controllerInstance = new PravegaControllerService("controller", zkUri);
+        controllerInstance = Utils.isDockerLocalExecEnabled()
```

Comment:
This line is repeated a lot. Perhaps it could be made into a facotory method.

Reply:
Done